src/HttpRequest.cpp

```diff
@@ -498,6 +498,32 @@ stringpair_t	HttpRequest::parseHeader(octets_t header)
 	return (authority);
 }
 
+void	HttpRequest::validateContentType(const Location &location)
+{
+	stringmap_t::iterator											contentType;
+	std::string														contentTypeValue;
+	std::map< std::string, std::set<std::string> >					mimeTypesDict;
+	struct stat														statBuff;
+
+	contentType = this->headerFields.find("content-type");
+	mimeTypesDict = location.getMimeTypes().getMimeTypesDict();
+	if (stat(this->targetResource.c_str(), &statBuff) != 0)
+		throw (ResponseException(500, "Internal Server Error"));
+	if (contentType != this->headerFields.end()) // content-type found in headerFields
+	{
+	// validate content-type header field
+		contentTypeValue = splitQuotedString(contentType->second, ';')[0];
+		contentTypeValue = trim(contentTypeValue);
+		std::transform(contentTypeValue.begin(), contentTypeValue.end(), contentTypeValue.begin(), tolower); // case-insensitive
+		std::cout << "CONTENT TYPE: " << contentTypeValue << std::endl;
+	    if (mimeTypesDict.find(contentTypeValue) == mimeTypesDict.end()) // content-type not found in mimeTypesDict
+		{
+			this->response.updateStatus(415, "Unsupported media type");
```

Comment:
The second argument provides details, not status explanation (that is encoded in statusDict already) - should say something like "Intended content type not supported by the server"